### PR TITLE
Support setting the URL of the JAR to resolve

### PR DESCRIPTION
**Pull request by Duhemm**
Previously, multiversion didn't have an option to manually force the URL
of a JAR to resolve, which made it harder for developers to test targets
with a custom built JAR.

This commit adds an attribute `url` to the `DependencyConfig` where the
URL of the JAR to resolve can be set. Whether multiversion should allow
`url` in the dependency configuration can be controller with
`--[no-]allow-url`.

Following the Coursier semantics of the `url=...` attribute,
dependencies that set the URL will be resolve intransitively.
Dependencies that set the URL but are not marked as intransitive will
trigger a warning message, and the resolution will be done
intransitively.

Dependencies that set the URL, but are evicted will trigger an error,
regardless of `--no-fail-on-evicted-declared`.

### CODE REVIEW

File: multiversion/src/main/scala/multiversion/commands/ExportCommand.scala [hunk 1]
```diff
@@ -320,6 +361,9 @@ case class ExportCommand(
    * `failOnEvictedDeclared` is set, then an error will be returned when such
    * dependencies are found.
    *
+   * When a declared dependency is evicted and sets the URL of the JAR to resole, then the
```
Comment (AngelaGuardia) [inline multiversion/src/main/scala/multiversion/commands/ExportCommand.scala:364]:
```suggestion
   * When a declared dependency is evicted and sets the URL of the JAR to resolve, then the
```

File: tests/src/test/scala/tests/commands/ExportCommandSuite.scala [hunk 2]
```diff
@@ -51,7 +51,7 @@ class ExportCommandSuite extends tests.BaseSuite with tests.ConfigSyntax {
          |  'org.slf4j:slf4j-log4j12:1.6.4' is a transitive dependency of slf4j-1.6.4.
          |  - dependency: org.slf4j:slf4j-log4j12:1.6.1
          |                ^
-         |warning: 1 declared dependency was evicted.
+         |warning: 1 warning(s) found.
```
Comment (AngelaGuardia) [inline tests/src/test/scala/tests/commands/ExportCommandSuite.scala:54]: This warning message seems less helpful now, will the user be getting a more informative message elsewhere?

Comment (eed3si9n) [reply tests/src/test/scala/tests/commands/ExportCommandSuite.scala:54]: Previously all warnings were eviction warnings, but now there might be either transitive URL warning or eviction warning. I think it's clear from the warning message itself.

**Comment by eed3si9n** [comment]
Does this impact the YAML output?

**Review by eed3si9n — APPROVED** [review]

**Comment by Duhemm** [comment]
> Does this impact the YAML output?

This shouldn't have an impact on the YAML report.

**Review by AngelaGuardia — APPROVED** [review]

**Pull request merged**